File: src/main/java/org/com/belog/global/error/ErrorMetadata.java
package org.com.belog.global.error;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.Instant;
import java.util.List;
import java.util.Objects;

@Schema(description = "오류 상세 정보")
public record ErrorMetadata(
        @Schema(description = "필드별 검증 오류 목록")
        List<FieldErrorDetail> fieldErrors,
        @Schema(description = "오류 발생 시각")
        Instant timestamp
) {

    public ErrorMetadata {
        fieldErrors = List.copyOf(fieldErrors);
    }

    public static ErrorMetadata empty() {
        return of(List.of());
    }

    public static ErrorMetadata of(List<FieldErrorDetail> fieldErrors) {
        Objects.requireNonNull(fieldErrors, "fieldErrors must not be null");
        return new ErrorMetadata(fieldErrors, Instant.now());
    }
}

File: src/main/kotlin/org/com/belog/global/error/ErrorMetadata.kt
package org.com.belog.global.error

import io.swagger.v3.oas.annotations.media.Schema
import java.time.Instant

@Schema(description = "오류 상세 정보")
@ConsistentCopyVisibility
data class ErrorMetadata private constructor(
    @field:Schema(description = "필드별 검증 오류 목록")
    val fieldErrors: List<FieldErrorDetail>,
    @field:Schema(description = "오류 발생 시각")
    val timestamp: Instant,
) {
    companion object {
        fun empty(): ErrorMetadata = of(emptyList())

        fun of(fieldErrors: List<FieldErrorDetail>): ErrorMetadata = ErrorMetadata(fieldErrors.toList(), Instant.now())
    }
}
